src/addEffectOnUnmount.tsx

```diff
@@ -0,0 +1,32 @@
+import {addEffect, UnchangedProps, flowMax, addHandlers} from 'ad-hok'
+
+import addPropTrackingRef from './addPropTrackingRef'
+import cleanupProps from './cleanupProps'
+
+type AddEffectOnUnmountType = <TProps>(
+  callback: (props: TProps) => () => void,
+) => UnchangedProps<TProps>
+
+const addEffectOnUnmount: AddEffectOnUnmountType = (callback) => {
+  const handlerName = 'addEffectOnUnmountHandler'
+  const handlerRefName = 'addEffectOnUnmountHandlerRef'
+  return flowMax(
+    addHandlers({
```

Comment:
The version of `addEffectOnUnmount()` from Water.org was currying over initial props (which isn't always what you want), so updated the implementation to run the unmount handler against fresh props